### PR TITLE
Add initial deployment workflow

### PR DESCRIPTION
**Pull request by kylelaker-ccpo**
This adds the initial deployment workflow. See https://github.com/dod-ccpo/atat-web-ui/runs/3025504228 for an example of what the deployment looks like.

### CODE REVIEW

**Comment by kylelaker-ccpo** [comment]
I am generally strongly opposed to keeping dead code as comments; however, these are going to be extremely useful for future iteration on these pipelines and we can't just pull them from history in the future since this is the only history where they exist. We need a really solid place to keep these commands and documentation if it's not acceptable to include them in this PR.

File: terraform/README.md [hunk 1]
```diff
@@ -11,3 +11,46 @@ The following resources must be created prior to running this Terraform:
  - a persistent storage account in which to store the state must exist
  - a sufficiently-permissioned role must exist and the credentials must be saved
    as secrets on this repository
+
+All subsequent steps are going to assume that these are both configured.
```
Comment (anhtran-ccpo) [inline terraform/README.md:15]: These steps only need to be ran one time ever right?

Comment (kylelaker-ccpo) [reply terraform/README.md:15]: I mean... sort of? You don't need to do it every time you deploy. But there are valid reasons to create different roles or use different storage accounts.

Comment (anhtran-ccpo) [inline terraform/README.md:15]: What are you referring to when you say "both are configured"? Is it `$AZURE_STORAGE_ACCOUNT` and `$AZURE_STORAGE_RESOURCE_GROUP`

Comment (anhtran-ccpo) [reply terraform/README.md:15]: Sorry I had deleted my comment before I knew you had responded. But the original question Kyle is answering is if these only need to be ran once.

Comment (kylelaker-ccpo) [reply terraform/README.md:15]: The first bullet maps to what later is called `AZURE_STORAGE_ACCOUNT`, yeah. The role is a separate thing. But I wanted to not define these terms up front. This section will need to be significantly reworked soon when we move past a POC. I didn't want to tightly couple those terms too much.

**Review by anhtran-ccpo — APPROVED** [review]

**Review by zachclark-ccpo — APPROVED** [review]

**Pull request merged**